### PR TITLE
Youtube download is now done in a pool

### PR DESCRIPTION
**Pull request by tkardozo**
issue #35 is now complete, using multiprocessing.Pool for now

### CODE REVIEW

File: flask_server/app/main/routes.py [hunk 1]
```diff
@@ -1,10 +1,29 @@
...
 
 @bp.route("/")
 @bp.route("/index")
 @login_required
 def index():
     return render_template("base.html")
+
+
+@bp.route("/youtube", methods=['GET', 'POST'])
+def youtube():
```
Comment (bdomingues) [inline flask_server/app/main/routes.py]: All this funcionality should be delegated to the YouTube class

File: flask_server/app/main/routes.py [hunk 2]
```diff
@@ -1,10 +1,29 @@
...
+    form = YoutubeForm(request.form)
+    if request.method == 'POST' and form.validate():
+        try:
+            yt = get_videos(str(form.url.data))
```
Comment (bdomingues) [inline flask_server/app/main/routes.py]: All this funcionality should be delegated to the YouTube class. Maybe?

File: flask_server/app/forms.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+from flask_wtf import FlaskForm
+from wtforms import StringField, SubmitField
+from wtforms.validators import DataRequired
+
+
+class YoutubeForm(FlaskForm):
+    url = StringField('Youtube URL',
```
Comment (bdomingues) [inline flask_server/app/forms.py]: What about creating a custom validator that uses a regex to validate the url?

**Comment by tkardozo** [comment]
suggestions addressed

File: flask_server/app/main/routes.py [hunk 4]
```diff
@@ -1,10 +1,20 @@
 from app.main import bp
-from flask import render_template
+from flask import render_template, flash, request
 from flask_login import login_required
-
-
+from app.forms import YoutubeForm
+from app.services.youtube.youtube import download
```
Comment (bdomingues) [inline flask_server/app/main/routes.py]: Function download should be renamed to something else - Example: YoutubeDownload - so it doesn't create conflicts later down the line when we have multiple streams to download from.

Alternative (maybe makes more sense), make sure download in on a  Youtube class?

**Pull request merged**